src/Tools/dotnet-monitor/CollectionRules/Actions/LoadProfilerAction.cs

```diff
@@ -0,0 +1,92 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Diagnostics.Monitoring.WebApi;
+using Microsoft.Diagnostics.NETCore.Client;
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Exceptions;
+using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Options.Actions;
+using Microsoft.Extensions.Logging;
+using System;
+using System.ComponentModel.DataAnnotations;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Actions
+{
+    internal sealed partial class LoadProfilerActionFactory :
+        ICollectionRuleActionFactory<LoadProfilerOptions>
+    {
+        private readonly IServiceProvider _serviceProvider;
+        private readonly ILogger<LoadProfilerActionFactory> _logger;
+
+        public LoadProfilerActionFactory(IServiceProvider serviceProvider, ILogger<LoadProfilerActionFactory> logger)
+        {
+            _serviceProvider = serviceProvider ?? throw new ArgumentNullException(nameof(serviceProvider));
+            _logger = logger ?? throw new ArgumentNullException(nameof(serviceProvider));
+        }
+
+        public ICollectionRuleAction Create(IEndpointInfo endpointInfo, LoadProfilerOptions options)
+        {
+            if (null == options)
+            {
+                throw new ArgumentNullException(nameof(options));
+            }
+
+            ValidationContext context = new(options, _serviceProvider, items: null);
+            Validator.ValidateObject(options, context, validateAllProperties: true);
+
+            return new LoadProfilerAction(_logger, endpointInfo, options);
+        }
+
+        internal sealed partial class LoadProfilerAction :
+            CollectionRuleActionBase<LoadProfilerOptions>
+        {
+            private readonly IEndpointInfo _endpointInfo;
+            private readonly ILogger _logger;
+            private readonly LoadProfilerOptions _options;
+
+            public LoadProfilerAction(ILogger logger, IEndpointInfo endpointInfo, LoadProfilerOptions options)
+                : base(endpointInfo, options)
+            {
+                _logger = logger ?? throw new ArgumentNullException(nameof(logger));
+                _endpointInfo = endpointInfo ?? throw new ArgumentNullException(nameof(endpointInfo));
+                _options = options ?? throw new ArgumentNullException(nameof(options));
+
+                if (string.IsNullOrEmpty(_options.Path))
```

Comment:
These IsNullOrEmpty and Guid.Empty checks are not necessary since the only caller is the outer class which already validates the options. Let's keep options validation consistent and in few places as possible.